contracts/mainnet/connectors/compound/v3/main.sol

```diff
@@ -0,0 +1,845 @@
+	 * @notice Deposit a token to Compound for lending / collaterization from a address and update the position of 'to'.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be supplied. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param from The address from where amount is to be supplied.
+	 * @param to The address on account of which the supply is made or whose positions are updated.
+	 * @param amt The amount of the token to deposit. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens deposited.
+	 */
+	function depositFromUsingManager(
+		address market,
+		address token,
+		address from,
+		address to,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+
+		require(
+			market != address(0) && token != address(0) && to != address(0),
+			"invalid market/token/to address"
+		);
+
+		bool isEth = token == ethAddr;
+		address token_ = isEth ? wethAddr : token;
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		if (token_ == getBaseToken(market)) {
+			require(
+				CometInterface(market).borrowBalanceOf(to) == 0,
+				"debt-not-repaid"
+			);
+		}
+
+		amt_ = setAmt(market, token_, from, amt_, isEth, ACTION.deposit);
+
+		CometInterface(market).supplyFrom(from, to, token_, amt_);
+		setUint(setId, amt_);
+
+		eventName_ = "LogDepositFromUsingManager(address,address,address,address,uint256,uint256,uint256)";
+		eventParam_ = abi.encode(market, token, from, to, amt_, getId, setId);
+	}
+
+	/**
+	 * @dev Withdraw base/collateral asset.
+	 * @notice Withdraw base token or deposited token from Compound.
+	 * @param market The address of the market.
+	 * @param token The address of the token to be withdrawn. (For ETH: 0xEeeeeEeeeEeEeeEeEeEeeEEEeeeeEeeeeeeeEEeE)
+	 * @param amt The amount of the token to withdraw. (For max: `uint256(-1)`)
+	 * @param getId ID to retrieve amt.
+	 * @param setId ID stores the amount of tokens withdrawn.
+	 */
+	function withdraw(
+		address market,
+		address token,
+		uint256 amt,
+		uint256 getId,
+		uint256 setId
+	)
+		public
+		payable
+		returns (string memory eventName_, bytes memory eventParam_)
+	{
+		uint256 amt_ = getUint(getId, amt);
+
+		require(
+			market != address(0) && token != address(0),
+			"invalid market/token address"
+		);
+
+		bool isEth = token == ethAddr;
+		address token_ = isEth ? wethAddr : token;
+
+		TokenInterface tokenContract = TokenInterface(token_);
+
+		uint256 initialBal = getAccountSupplyBalanceOfAsset(
+			address(this),
+			market,
+			token_
+		);
+
+		amt_ = amt_ == uint256(-1) ? initialBal : amt_;
+
+		if (token_ == getBaseToken(market)) {
+			uint256 balance = CometInterface(market).balanceOf(address(this));
+			if (balance > 0) {
+				require(amt_ <= balance, "withdraw-amt-greater-than-supplies");
+			}
```

Comment:
@Richa-iitr If balanceOf is `0` it means the user could have borrowed base token. I think this condition doesn't satisfy the required check. 

You need to check borrowBalance here. And if that is greater than `0`. It should revert.

Reply:
added both conditions